File: Src/Icm.Core.Tests/Basic types extensions/DoubleExtensionsTest.vb
<Category("Icm")>
<TestFixture()>
Public Class DoubleExtensionsTest

    <TestCase(5.2348R, 1, Result:=5.2R)>
    Public Function ChangePrecision_Test(target As Double, precision As Integer) As Double
        Return target.ChangePrecision(precision)
    End Function

    <TestCase(90.0R, Result:=Math.PI / 2)>
    <TestCase(0.0R, Result:=0)>
    Public Function Deg2Rad_Test(target As Double) As Double
        Return target.Deg2Rad
    End Function

    <TestCase(Math.PI / 2, Result:=90.0R)>
    <TestCase(0.0R, Result:=0.0R)>
    Public Function Rad2Deg_Test(target As Double) As Double
        Return target.Rad2Deg
    End Function

End Class

File: Src/Icm.Core.Tests/Basic types extensions/DoubleExtensionsTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm;
using NUnit.Framework;

[Category("Icm")]
[TestFixture()]
public class DoubleExtensionsTest
{

	[TestCase(5.2348, 1, ExpectedResult = 5.2)]
	public double ChangePrecision_Test(double target, int precision)
	{
		return target.ChangePrecision(precision);
	}

	[TestCase(90.0, ExpectedResult = Math.PI / 2)]
	[TestCase(0.0, ExpectedResult = 0)]
	public double Deg2Rad_Test(double target)
	{
		return target.Deg2Rad();
	}

	[TestCase(Math.PI / 2, ExpectedResult = 90.0)]
	[TestCase(0.0, ExpectedResult = 0.0)]
	public double Rad2Deg_Test(double target)
	{
		return target.Rad2Deg();
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
